flake.nix

```diff
@@ -0,0 +1,49 @@
+{
+  description = "Python environment for fipy";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
```

Comment:
Maintenance overhead question: @wd15, do you plan to update this URL each May and November?

Reply:
It doesn't need to be updated necessarily, just when we feel like it. It's a single place to update rather than multiple places so in that sense it's good. Can also pin to the latest release so whenever "nix flake update" is executed it gets the latest versions, but that is prone to failure if it's not tested (so still sort of protected from updates even with latest). I'm happy to switch it to latest, but I've been using a specific release for my projects.